src/Catalog/Registration/RegistrationCollector.cs

```diff
@@ -131,5 +281,155 @@ public static ShouldIncludeRegistrationPackage GetShouldIncludeRegistrationPacka
 
             return (k, u, g) => !NuGetVersionUtility.IsGraphSemVer2(k.Version, u, g);
         }
+
+        private static void DequeueBatches(Queue<BatchTask> batches, Func<BatchTask, bool> isMatch)
```

Comment:
Here you dequeue batches only while the front matches. Consider changing the name to DequeueBatchesWhileMatches or something else that explains that